docs/design/queue/queue-suspend.md

```diff
@@ -0,0 +1,176 @@
+
+### Story 2
+
+<!--
+
+作为一个用户，我希望集群管理员在对队列进行运维操作时，我可以继续进行作业的提交，不应该对提交作业进行限制。
+
+-->
+
+As a user, I hope that when the cluster administrator performs operation and maintenance operations to the queue, I can continue to submit jobs, and there should be no restrictions on job submissions.
+
+### Story 3
+
+<!--
+
+作为一个集群管理员，在对队列的运维操作结束后，我希望可以恢复队列的调度，在队列运维过程中提交的作业运行可以恢复调度，新提交的作业也可以调度。
+
+-->
+
+As a cluster administrator, after the operation and maintenance of the queue is completed, I hope that the scheduling of the queue can be restored. The jobs submitted during the queue operation and maintenance can be resumed and the newly submitted jobs can also be scheduled.
+
+## Implementation
+
+<!--
+
+我们需要修改volcano的以下几个部分来支持队列的暂停/恢复调度能力。
+
+-->
+
+We need to modify the following parts of volcano to support the queue's pause/resume scheduling capabilities.
+
+### API Change
+
+<!--
+
+在pkg/apis/scheduling/v1beta1/types.go文件中为队列增加一个新的状态`Suspend`，表示队列处于暂停中。
+
+在pkg/apis/bus/v1alpha1/actions.go文件中新增两个action，SuspendQueue和ResumeQueue，分别表示暂停队列和恢复队列调度的动作。
+
+-->
+
+Add a new status `Suspend` to the queue in the pkg/apis/scheduling/v1beta1/types.go, indicating that the queue is suspended.
+
+```go
+// QueueStateOpen indicate `Open` state of queue
+QueueStateOpen QueueState = "Open"
+// QueueStateClosed indicate `Closed` state of queue
+QueueStateClosed QueueState = "Closed"
+// QueueStateClosing indicate `Closing` state of queue
+QueueStateClosing QueueState = "Closing"
+// QueueStateUnknown indicate `Unknown` state of queue
+QueueStateUnknown QueueState = "Unknown"
+// QueueStateSuspended indicate `Suspended` state of queue
+QueueStateSuspended QueueState = "Suspended" # New added
+```
+
+Add two new actions in the pkg/apis/bus/v1alpha1/actions.go, `Suspend` and `Resume`, which represent the actions of suspending the queue and resuming queue scheduling respectively.
+
+```go
+// OpenQueueAction is the action to open queue
+OpenQueueAction Action = "OpenQueue"
+
+// CloseQueueAction is the action to close queue
+CloseQueueAction Action = "CloseQueue"
+
+// SuspendQueueAction is the action to suspend queue
+SuspendQueueAction Action = "SuspendQueue"
+
+// ResumeQueueAction is the action to resume queue
+ResumeQueueAction Action = "ResumeQueue"
+```
+
+
+
+### Volcano components
+
+#### Volcano controller
+
+<!--
+
+在pkg/controllers/queue/state中新增suspended文件，处理Open，Resume，Close action，根据action类型分别更新队列到相应的状态。
+
+队列在Open，Closing，Unknown状态下时，处理Suspend和Resume action。
+
+- 当队列为Open，Closing或Unknown状态时，收到Suspend action，将队列更新为suspended状态，收到Resume action时，不做任何处理。
```

Comment:
I see in `checkQueueState()` that only `Open` queues can Submit jobs now.
Is it expected that a Queue in the `Unknown` state should not accept Job submissions?
If so, should accepting the `Suspended` state be considered as inconsistent with the job submission behavior in the `Unknown` state, as is the case with `Closed`?

Reply:
Yeah `checkQueueState()` block pod submission，pod&job can be submitted when queue is `open` or `suspended`.